pkg/operation/common/utils.go

```diff
@@ -427,61 +434,52 @@ func InjectCSIFeatureGates(kubeVersion string, featureGates map[string]bool) (ma
+		return errors.New("must provide non-nil kubernetes client to common.DeleteLoggingStack")
 	}
 
-	var (
-		services     = []string{"kibana-logging", "elasticsearch-logging", "fluentd-es"}
-		configmaps   = []string{"kibana-object-registration", "kibana-saved-objects", "curator-hourly-config", "curator-daily-config", "fluent-bit-config", "fluentd-es-config", "es-configmap"}
-		statefulsets = []string{"elasticsearch-logging", "fluentd-es"}
-		cronjobs     = []string{"hourly-curator", "daily-curator"}
-	)
+	// TODO: Use `DeleteCollection` as soon it is in the controller-runtime:
+	// https://github.com/kubernetes-sigs/controller-runtime/pull/324
 
-	if err := k8sClient.DeleteDeployment(namespace, "kibana-logging"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := k8sClient.DeleteDaemonSet(namespace, "fluent-bit"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	for _, name := range statefulsets {
-		if err := k8sClient.DeleteStatefulSet(namespace, name); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-	for _, name := range cronjobs {
-		if err := k8sClient.DeleteCronJob(namespace, name); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
+	// Delete the resources below that match "garden.sapcloud.io/role=logging"
+	lists := []runtime.Object{
+		&corev1.ConfigMapList{},
+		&batchv1beta1.CronJobList{},
+		&rbacv1.ClusterRoleList{},
+		&rbacv1.ClusterRoleBindingList{},
+		&appsv1.DaemonSetList{},
+		&appsv1.DeploymentList{},
+		&autoscalingv1.HorizontalPodAutoscalerList{},
+		&extensionsv1beta1.IngressList{},
+		&corev1.SecretList{},
+		&corev1.ServiceAccountList{},
+		&corev1.ServiceList{},
+		&appsv1.StatefulSetList{},
 	}
 
-	if err := k8sClient.DeleteIngress(namespace, "kibana"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := k8sClient.DeleteSecret(namespace, "kibana-basic-auth"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := k8sClient.DeleteClusterRoleBinding("fluent-bit-read"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := k8sClient.DeleteClusterRole("fluent-bit-read"); err != nil && !apierrors.IsNotFound(err) {
-		return err
-	}
-	if err := k8sClient.DeleteServiceAccount(namespace, "fluent-bit"); err != nil && !apierrors.IsNotFound(err) {
+	requirement, err := labels.NewRequirement(GardenRole, selection.Equals, []string{GardenRoleLogging})
+	if err != nil {
 		return err
 	}
-	if err := k8sClient.DeleteHorizontalPodAutoscaler(namespace, "fluentd-es"); err != nil && !apierrors.IsNotFound(err) {
-		return err
+	listOptions := &client.ListOptions{
+		Namespace:     namespace,
+		LabelSelector: labels.NewSelector().Add(*requirement),
 	}
-	for _, name := range services {
-		if err := k8sClient.DeleteService(namespace, name); err != nil && !apierrors.IsNotFound(err) {
+
+	for _, list := range lists {
+		if err := k8sClient.Client().List(context.TODO(), listOptions, list); err != nil {
```

Comment:
Please add a TODO here that we should switch to `DeleteCollection` once it's finally there kubernetes-sigs/controller-runtime#324